core/cache/controller.go

```diff
@@ -173,6 +186,7 @@ func (c *Controller) loop() error {
 // Mark updates all cached entities to indicate they are stale.
 func (c *Controller) Mark() {
 	c.manager.mark()
+	c.initializing = true
```

Comment:
Looks like Mark and Sweep are called on different go routines. (I'm reading the modelcache worker). So this should probably be behind a mutex or something else.

Reply:
No, Mark and Sweep are called on the same goroutine.

Reply:
Actually, you're right... I'm thinking...

Reply:
Added locking.